tests/functional/test_pkg_resources_provider.py

```diff
@@ -0,0 +1,91 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2005-2020, PyInstaller Development Team.
+#
+# Distributed under the terms of the GNU General Public License (version 2
+# or later) with exception for distributing the bootloader.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#
+# SPDX-License-Identifier: (GPL-2.0-or-later WITH Bootloader-exception)
+#-----------------------------------------------------------------------------
+
+
+# Library imports
+# ---------------
+import copy
+import os
+
+# Third-party imports
+# -------------------
+import pytest
+
+# Local imports
+# -------------
+from PyInstaller.utils.tests import importorskip
+from PyInstaller.compat import exec_python_rc
+
+# Directory with testing modules used in some tests.
+_MODULES_DIR = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)),
+    'modules'
+)
+
+
+def __exec_python_script(script_filename, *args, pathex=None):
+    cmd = [script_filename]
+    cmd.extend(args)
+
+    # Prepare the environment - default to 'os.environ'...
+    env = copy.deepcopy(os.environ)
+    # ... and prepend PYTHONPATH with pathex
+    if pathex:
+        if 'PYTHONPATH' in env:
+            pathex = os.pathsep.join([pathex, env.get('PYTHONPATH')])
```

Comment:
```suggestion
            pathex = os.pathsep.join([pathex, env.['PYTHONPATH']])
```
is already known to exist, so we ban safely use item-access.

Reply:
Indeed. Will be changed in next revision.